codegen/module.go

```diff
@@ -104,6 +90,79 @@ func (system *ModuleSystem) RegisterClass(class ModuleClass) error {
 	return nil
 }
 
+// dedup returns a slice containing unique sorted elements of the given slice
+func dedup(array []string) []string {
+	dict := map[string]bool{}
+	for _, elt := range array {
+		dict[elt] = true
+	}
+
+	i := 0
+	unique := make([]string, len(dict))
+	for key := range dict {
+		unique[i] = key
+		i++
+	}
+	sort.Strings(unique)
+	return unique
+}
+
+// validateDir checks if the module class can map to the given dir
+func (system *ModuleSystem) validateClassDir(class *ModuleClass, dir string) error {
+	dir = filepath.Clean(dir)
```

Comment:
This clean step is part of the validation which is used at multiple places and does not assume the given dir is clean. 

The other clean is before dedup, which is before validation. So yes, we cleaned twice in `RegisterClass`, but it doesn't hurt.